### PR TITLE
✨(project) tool base with a Docker compose configuration

### PR DESCRIPTION
**Pull request by quitterie-lcs**
# Purpose

The aim of this tool is to constitute a dataset of learning logs from any existing LMS. In its first implementation, Openedx is the LMS used. To get and store logs from the LMS chosen by the user, a graylog server is deployed.

# Proposal

- [x] add the `docker-compose` file to settle the graylog environment
- [x] add user documentation (in `README.md`)

### CODE REVIEW

File: docker-compose.edx.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            - edx_mailcatcher
+            - edx_mysql
+            - edx_mongodb
+            - edx_memcached
+            - edx_redis
+        networks:
+            - default    
+
+    dockerize:
+        image: jwilder/dockerize
```
Comment (jmaupetit) [inline docker-compose.edx.yml:55]: A carriage return is missing here. You can configure your editor to do so when saving edited file.

File: docker-compose.edx.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    edx_redis:
+        image: redis:4.0.10-alpine
+    
+    edx_mailcatcher:
+        image: sj26/mailcatcher:latest
+        ports:
+            - "1080:1080"
+
+    edx_lms:
+        image: fundocker/edxapp:dogwood.3-fun-2.1.1
```
Comment (jmaupetit) [inline docker-compose.edx.yml:31]:
```suggestion
        image: ${EDXAPP_IMAGE:-fundocker/edxapp:dogwood.3-fun-2.1.1}
```

File: docker-compose.edx.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            DJANGO_SETTINGS_MODULE: lms.envs.fun.docker_run
+        ports:
+            - "8072:8000"
+        volumes:
+            - ./releases/dogwood/3/fun/data/media:/edx/var/edxapp/media
+            - ./releases/dogwood/3/fun/config:/config
+        command: >
+            python manage.py lms runserver 0.0.0.0:8000
+        depends_on:
+            - graylog
```
Comment (jmaupetit) [inline docker-compose.edx.yml:45]: I would not add this dependency as the graylog service is describe in another configuration file.

File: env.d/edx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+EMAIL_PORT=1025
+
+# Python
+PYTHONUNBUFFERED=1
+
+# Features
+FEATURES={"AUTOMATIC_AUTH_FOR_TESTING": true, "RESTRICT_AUTOMATIC_AUTH": false, "ENABLE_OAUTH2_PROVIDER": true, "ENABLE_CORS_HEADERS": true, "ENABLE_CROSS_DOMAIN_CSRF_COOKIE": true}
+OAUTH_ENFORCE_SECURE=
+OAUTH_ENFORCE_CLIENT_SECURE=
+OAUTH_OIDC_ISSUER=http://edx:8073/oauth2
```
Comment (jmaupetit) [inline env.d/edx:25]:
```suggestion
OAUTH_OIDC_ISSUER=http://edx_lms:8000/oauth2
```

File: env.d/edx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+SESSION_COOKIE_NAME=edxcsrftoken
+# Do not enable CORS_ORIGIN_ALLOW_ALL for production purpose!
+# Instead set CORS_ORIGIN_WHITELIST properly
+CORS_ORIGIN_ALLOW_ALL=true
+
+# API
+EDX_API_KEY=FakeEdXAPIKey
+
+# Richie synchronization
+COURSE_HOOKS=[{"url": "https://richie:8070/api/v1.0/course-runs-sync/", "secret": "shared secret", "verify": false}]
```
Comment (jmaupetit) [inline env.d/edx:41]: We can remove this?

File: releases/dogwood/3/fun/entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/entrypoint.sh:1]: You can remove this file.

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+            - default
+        depends_on:
+            - mongo
+            - elasticsearch
+        ports:
+            - 9000:9000
+            - 12201:12201
+    
+    dockerize:
+        image: jwilder/dockerize
```
Comment (jmaupetit) [inline docker-compose.yml:42]: Ditto (carriage return).

File: README.md [hunk 8]
```diff
@@ -1,5 +1,70 @@
...
+
+## Starting with the tool
+
+1. Clone the current repository.
+
+2. To activate settings for the `dogwood` release used in the project, run the following command:
+
+    ```bin/activate```
+
+3. Follow the written instructions.
```
Comment (jmaupetit) [inline README.md:18]: Outdated?

File: README.md [hunk 9]
```diff
@@ -1,5 +1,70 @@
 # Learning analytics playground
 
+## Description
+This tool is a first draft of connecting an LMS interface to a Graylog server to create a dataset 
+of learning logs.
```
Comment (jmaupetit) [inline README.md:5]:
```suggestion
## Description

This tool is a first draft of connecting an LMS interface to a Graylog server to create datasets
of learning logs.
```

File: README.md [hunk 10]
```diff
@@ -1,5 +1,70 @@
...
+It is intended to be used locally. 
+Currently, it is using LMS from OpenEdx and the FUN's release of dogwood.
+
+## Starting with the tool
+
+1. Clone the current repository.
+
+2. To activate settings for the `dogwood` release used in the project, run the following command:
+
+    ```bin/activate```
```
Comment (jmaupetit) [inline README.md:16]: Code can be written without indentation but using triple back quotes notation :

```suggestion
'''
bin/activate
'''
```

__nota bene__: the previous suggestion does not have back quotes but quotes, 'cause I cannot escape them :stuck_out_tongue:

File: README.md [hunk 11]
```diff
@@ -1,5 +1,69 @@
 # Learning analytics playground
 
+## Description
+
+This tool is a first draft of connecting an LMS interface to a Graylog server to create a dataset 
+of learning logs.
+
+It is intended to be used locally. 
+Currently, it is using LMS from OpenEdx and the FUN's release of dogwood.
```
Comment (jmaupetit) [inline README.md:9]: Maybe add a link to openfun/openedx-docker?

```suggestion
Currently, it is using OpenEdx's LMS service (with the fun-flavored dogwood's release).
```

File: README.md [hunk 12]
```diff
@@ -1,5 +1,69 @@
...
+
+## Starting with the tool
```
Comment (jmaupetit) [inline README.md:11]:
```suggestion
## Getting started
```

File: README.md [hunk 13]
```diff
@@ -1,5 +1,69 @@
...
+
+- When using the tool for the first time, you need to run migrations. Run the following command:
+
+```
+make migrate
+```
+
+The tool is now ready for use ! :rocket:
+
+## Use
```
Comment (jmaupetit) [inline README.md:23]:
```suggestion
## Usage
```

File: README.md [hunk 14]
```diff
@@ -1,5 +1,69 @@
...
+1. Run the following command:
+
+```
+make run
+```
+
+2. When connections are established, you can access:
+
+    * the LMS server on `localhost:8072`
+    * the Graylog server on `127.0.0.1:9000`
```
Comment (jmaupetit) [inline README.md:38]:
```suggestion
    * the LMS server on [localhost:8072](http://localhost:8072)
    * the Graylog server on [localhost:9000](http://localhost:9000)
```

File: README.md [hunk 15]
```diff
@@ -1,5 +1,69 @@
...
+
+### Configure an input on Graylog interface
+
+1. Access Graylog interface with admin/admin credentials.
```
Comment (jmaupetit) [inline README.md:42]:
```suggestion
1. Access Graylog interface with `admin` / `admin` credentials.
```

File: Makefile [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+	$(COMPOSE) rm edx_mongodb edx_mysql edx_redis 
+.PHONY: clean-db
+
+logs:  ## get development logs
+	$(COMPOSE) logs -f
+.PHONY: logs
+
+# Nota bene: we do not use the MANAGE_* shortcut because, for some releases
+# (e.g.  dogwood), we cannot run the LMS while migrations haven't been
+# performed.
```
Comment (jmaupetit) [inline Makefile:49]: Copy/pasta to remove.

File: Makefile [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+
+# Target release expected tree
+
+data/edx/media/.keep:
+	mkdir -p data/edx/media
+	touch data/edx/media/.keep
+
+# Make commands
+
+clean:  ## remove downloaded sources
```
Comment (jmaupetit) [inline Makefile:32]:
```suggestion
clean:  ## remove temporary data
```

File: Makefile [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+	rm -r \
+	  data/* || exit 0
+.PHONY: clean
+
+clean-db: \
+  stop
+clean-db:  ## remove mongo, mysql & redis databases
```
Comment (jmaupetit) [inline Makefile:39]:
```suggestion
clean-db:  ## remove LMS databases
```

**Review by jmaupetit — APPROVED** [review]
I only have minor remarks.

File: README.md (hunk 11 above)
Comment (quitterie-lcs) [reply README.md:9]: Good idea

**Pull request merged**